### PR TITLE
Fix tests for new profile edit UI

### PR DESCRIPTION
**Pull request by comzeradd**
Worked on top of @bebef1987 commit. Fixed some minor issues and registration tests.

Ideally we should have more tests for the Settings page, to cover all tabs and subforms, but it's a good start. We may also need some further clean up for methods that are no longer needed for the UI, but these call happen in another PR.

All tests pass locally.

### CODE REVIEW

**Comment by johngian** [comment]
Hey @m8ttyB! Can you take a look at this?

**Comment by justinpotts** [comment]
@johngian @comzeradd: Tests look good to me, but are failing locally. Then again, I had some trouble getting some files set up in my local repo so it'd be worth having @m8ttyB run these to double check. I'll also go over my repo settings again and see if I can get them to pass this evening.

Once they run locally, either for me or @m8ttyB, r+.

**Comment by bebef1987** [comment]
All looks OK but did not managed to run the tests.

**Comment by davehunt** [comment]
@comzeradd do we expect these to only pass on dev? If so, when we merge this we'll have failures for staging and production test suites. Do we want to disable these suites, or skip tests based on the target environment?

File: tests/test_register.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+from unittestzero import Assert
+from pages.home_page import Home
+
+
+class TestRegister:
+
+        def test_profile_creation(self, base_url, selenium, new_user):
```
Comment (bebef1987) [inline tests/test_register.py]: I think the spacing is off here 
one <tab> = 4 spaces

**Comment by bebef1987** [comment]
Ran the tests and they are passing
 I will let you guys decide when we should merge these tests

**Comment by AndreiH** [comment]
Tested this on DEV environment and tests are good.
Didn't had a chance to try this on STAGE, because I need a vouched account.
If the environments are different, @davehunt may I suggest running from different branches?

**Comment by davehunt** [comment]
Running from different branches works, but introduces a workflow that will require maintenance. There's currently just one branch, and CI configuration would also need to be updated. Note that I don't think any of our other suites (which likely have more resources that Mozillians) use branches and when we have in the past there have been issues with keeping the branches up to date.

**Comment by comzeradd** [comment]
Right now this only passes on the dev instance of mozillians, since all the UI changes regarding profile/registration are not yet pushed to either stage or prod. I don't know how you usually handle this when things change and not all tests pass on all environments.

**Comment by bebef1987** [comment]
I think the easiest way to do this is xfail the tests on stage and prod

**Comment by comzeradd** [comment]
@bebef1987 @davehunt I marked the tests that fail either on stage or production.

**Comment by johngian** [comment]
@davehunt If ``xfail`` is the way to go, is this ready to be merged?

File: tests/test_profile.py [hunk 2]
```diff
@@ -15,153 +15,101 @@
 
 class TestProfile:
 
+    @pytest.mark.xfail("'mozillians.allizom','mozillians.org' in config.getvalue('base_url')",
```
Comment (davehunt) [inline tests/test_profile.py]: This would return `('mozillians.allizom', False)` and so is not doing what you're expecting.

**Comment by davehunt** [comment]
> @davehunt If `xfail` is the way to go, is this ready to be merged?

I have not reviewed this patch, so I can't say if it's ready to be merged. The xfail used does not appear to be doing what you're expecting though. You want to xfail if the base URL contains either one of two strings, but the syntax you've used will not do that.

**Comment by comzeradd** [comment]
@davehunt PR updated. It's only two tests that are expected to fail on production, so I used an extra decorator just to be safe. Could you give it another pass?

**Comment by comzeradd** [comment]
@justinpotts @m8ttyB PR updated. All tests pass locally for dev. Those that fail on prod or stage, due to UI changes not yet deployed there, are marked xfail.

**Comment by m8ttyB** [comment]
Adding @mozilla/web-qa-sorcerers

**Comment by stephendonner** [comment]
Carrying forward @bebef1987 and @AndreiH's r+d reviews.

I'm taking the liberty of merging this in to restore much-needed _central_ CI coverage for the team, as they work to complete the changes necessitated by Legal et al by the end of the year.

Here are the runs from staging Jenkins (http://webqa-ci-staging1.qa.scl3.mozilla.com:8080/ - need Mozilla VPN and access) with this pull:

1) prod: https://pastebin.mozilla.org/8855253
2) staging: https://pastebin.mozilla.org/8855252
3) dev: https://pastebin.mozilla.org/8855251

I've re-run these a few times each, without issue, but it's possible we'll see timing/instability issues, going forward; so from here on out, we should be careful to:

1) monitor builds and respond to failures/timeouts, etc. with bugs/app fixes/test fixes
2) continue to review and refine the tests, across the 3 run environments

Thanks for the work to get us to this point, @bebef1987 and @comzeradd - can you please work with the team to help ensure the suites stay stable and bugs are tracked, etc.?

**Pull request merged**